Libraries/LibWeb/HTML/PopoverInvokerElement.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024, Nathan van der Kamp <nbvdkamp@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGC/Ptr.h>
+#include <LibWeb/Forward.h>
+
+namespace Web::HTML {
+
+class PopoverInvokerElement {
+public:
+    PopoverInvokerElement() { }
+
+    GC::Ptr<DOM::Element> get_popover_target_element() { return m_popover_target_element; }
+
+    void set_popover_target_element(GC::Ptr<DOM::Element> value) { m_popover_target_element = value; }
+
+private:
+    GC::Ptr<DOM::Element> m_popover_target_element;
```

Comment:
This needs to be visited by a `visit_edges()` method.

Reply:
When trying to apply the GC visit_edges pattern I'm seeing elsewhere by making PopoverInvokerElement inherit from `JS:Cell` I run into the problem that `HTMLButtonElement` is then inheriting `Cell` from two different base classes (PopoverInvokerElement  and Object). This gives the following errors:
```
LibJS/Runtime/Realm.h:39:9: error: ‘JS::Cell’ is an ambiguous base of ‘Web::HTML::HTMLButtonElement’
   39 |         static_cast<Cell*>(object)->initialize(*this);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
LibGC/Heap.h:47:17: error: ‘GC::Cell’ is an ambiguous base of ‘Web::HTML::HTMLButtonElement’
   47 |         return *static_cast<T*>(memory);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~
```
Is there an established way of getting around this?